Update referenced initiatives for power grids mapping

ekatef opened the pull request:
PR aims to highlight the existing opportunities for energy modeles to use and contribute into power grid mapping

ekatef committed:
Update sources on power grids

ekatef commented:
@davide-f @hazemakhalek could you please let me know if you have any comments on this PR? It's needed to clarify links with the OSM mapping initiative.

davide-f reviewed (commented):
Is there a reason to remove the other one? Maybe the fact that dynartio is part of mapyourgrid? 

I don't see any harm in leaving the previous line in case; whatever you decide is fine by me

ekatef commented:
> Is there a reason to remove the other one? Maybe the fact that dynartio is part of mapyourgrid?
> 
> I don't see any harm in leaving the previous line in case; whatever you decide is fine by me

@davide-f many thanks for checking! I think it's best to keep focus in the description of the initiative and place link to strategic projects. The visualisation tool is more instrumental and a link to it should be better placed in the documentation. Have added a link to it to [the PR](https://github.com/pypsa-meets-earth/pypsa-earth/pull/1561) with the validation hints.

ekatef commented:
If there is no further comments I'll merge it?

ekatef committed:
Add a link to Awesome Electricity Demand

ekatef committed:
Update explanations

ekatef commented:
Have added an issue https://github.com/pypsa-meets-earth/pypsa-earth/issues/1639 to remember about keeping a link with the visualisation toolkit. Have also added a link to Awesome Electricity Demand collection which has the same spirit as `Awesome Electrical Grid Mapping` collection.

Merging the PR since it's needed to keep the description up-to-date to be revised and updated later.
